### PR TITLE
fix for Django 2.0, update tests

### PR DESCRIPTION
**Pull request by PetrDlouhy**
Fixes for Django 2.0

### CODE REVIEW

**Comment by asfaltboy** [comment]
Thanks @PetrDlouhy that's mighty kind of you! I think Travis must use some cached mirror doesn't have [Django 2.0.1][1] on it yet.

[1]: https://pypi.python.org/pypi/Django/2.0.1

**Comment by PetrDlouhy** [comment]
@asfaltboy No, Django 2.0 doesn't support Python 2, that is cause for the fail of the first two builds.
The cause of the Python 3 build is that there will be needed more compatibility changes.

I will resolve those problems.

File: advanced_filters/models.py [hunk 1]
```diff
@@ -19,8 +19,12 @@ class Meta:
         verbose_name_plural = _('Advanced Filters')
 
     title = models.CharField(max_length=255, null=False, blank=False, verbose_name=_('Title'))
-    created_by = models.ForeignKey(settings.AUTH_USER_MODEL,
-                                   related_name='created_advanced_filters', verbose_name=_('Created by'))
+    created_by = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        related_name='created_advanced_filters',
+        verbose_name=_('Created by'),
+        on_delete=models.CASCADE,
```
Comment (asfaltboy) [inline advanced_filters/models.py]: This is fine right now, but may need to be altered if we implement advanced filter sharing.

File: tox.ini [hunk 2]
```diff
@@ -1,11 +1,11 @@
 [tox]
 envlist =
     py26-d{15,16}
-    py27-d{15,16,17,18,19,110}
+    py27-d{15,16,17,18,19,110,20}
```
Comment (asfaltboy) [inline tox.ini]: Shall we remove 2.0 from py27 env in tox as well?

File: advanced_filters/models.py (hunk 1 above)
Comment (asfaltboy) [inline advanced_filters/models.py]: This is fine right now, but may need to be altered if we implement AdvancedFilter sharing.

**Comment by PetrDlouhy** [comment]
@asfaltboy I fixed all the tests. The failing tests are due to `django-braces` which are not yet released for Django 2.0.

**Comment by cyberbudy** [comment]
Any progress on it?

File: advanced_filters/tests/test_admin.py [hunk 3]
```diff
@@ -1,4 +1,7 @@
-from django.core.urlresolvers import reverse
+try:
+    from django.urls import reverse
+except:  # Django < 2.0
```
Comment (asfaltboy) [inline advanced_filters/tests/test_admin.py]: I prefer using `except ImportError` in such cases

**Comment by asfaltboy** [comment]
Sorry @cyberbudy , we're sort of blocked by 3rd party package compatibility issue (brack3t/django-braces#233).

You may still use PetrDlouhy's `develop` branch try to workaround the issue by providing your own copy of the mixins instead of braces' ones (that won't fail in 2.0), as here:

https://github.com/modlinltd/django-advanced-filters/blob/1a482cf85709d28adefe6b4444eac8043eed701b/advanced_filters/views.py#L21-L22

**Comment by lnxg33k** [comment]
django-braces pumped the code to cover django 2 https://github.com/brack3t/django-braces/commit/8b2f4122ea02f907ad9e9e29fa19d90f594ecebb

**Comment by asfaltboy** [comment]
Seems like some `Q()` object serialisation assertions are actually failing now [on Python 3.5 w/ Django 2 in Travis][1]. We'll need to have a look at this before we merge, help is appreciated!

[1]: https://travis-ci.org/modlinltd/django-advanced-filters/jobs/379085311

**Comment by mrname** [comment]
Might want to add a database migration in this PR. I just ran `makemigrations` from your branch, and got some updates:

```
Migrations for 'advanced_filters':
  /usr/local/lib/python3.6/site-packages/advanced_filters/migrations/0003_auto_20180606_1855.py
    - Alter field created_at on advancedfilter
    - Alter field created_by on advancedfilter
    - Alter field groups on advancedfilter
    - Alter field title on advancedfilter
    - Alter field url on advancedfilter
    - Alter field users on advancedfilter
```

**Comment by peppelinux** [comment]
This may help, actually I'm using this in Django 2.0:
https://github.com/modlinltd/django-advanced-filters/pull/73

**Comment by anuj9196** [comment]
Is there any update for `Dajngo 2.0`?

**Comment by mrsarm** [comment]
When will this merge? I tested the branch proposed in my project with Django 2.0.7 and Python 3.6 and it worked!

For those that need this asap, add this to your `requirements.txt`:

    -e git://github.com/PetrDlouhy/django-advanced-filters.git@e769781ac489dfd408c2f872b2d60603451fc677#egg=django-advanced-filters

**Comment by PetrDlouhy** [comment]
@asfaltboy I looked at the changes, and they look good to me. Thanks for the `except` fix, that was oversight from my side.

There is one think, I thing is not right (but was not right before this change). And it is the `install_requires` in `setup.py`. I think, we shouldn't stick to one version. It could lead to unsolvable dependencies problem. I think, we should change it to:
```
    install_requires=[
        'django-braces>=1.4.0',
        'simplejson>=3.6.5',
    ],
```

**Comment by asfaltboy** [comment]
Thanks for the review Petr! Together with PR #70 (regarding missing migrations) this will make for a nice small "support release".

Regarding the dependency version changes, it was discussed in #79 and @predatell has convinced me that this change is for the best for libraries (to allow users more flexibility within reason). Please have a look at that ticket's comments and feel free to add your own comments there.

**Pull request merged**